app/tests/spec/lib/app-start.js

```diff
@@ -150,6 +160,142 @@ function (chai, sinon, AppStart, Session, Constants, p,
           });
       });
     });
+
+    describe('initializeAuthenticationBroker', function () {
+      beforeEach(function () {
+        appStart = new AppStart({
+          window: windowMock,
+          router: routerMock,
+          history: historyMock,
+          user: userMock
+        });
+      });
+
+      describe('fx-desktop', function () {
+        it('returns an FxDesktop broker if `service=sync`', function () {
+          windowMock.location.search = Url.objToSearchString({
+            service: Constants.FX_DESKTOP_SYNC
+          });
+          return appStart.initializeAuthenticationBroker()
```

Comment:
There is a lot of boilerplate here. A helper function that accepts the expected broker could be used to reduce LOC.